lib/miner-info.js

```diff
@@ -1,140 +1,33 @@
+import {
+  getIndexProviderPeerId as getPeerId,
+  MINER_TO_PEERID_CONTRACT_ADDRESS,
+  MINER_TO_PEERID_CONTRACT_ABI,
+  ethers,
+} from '../vendor/deno-deps.js'
+
+// Initialize contract
+const fetchRequest = new ethers.FetchRequest(RPC_URL)
+fetchRequest.setHeader('Authorization', `Bearer ${RPC_AUTH}`)
+const provider = new ethers.JsonRpcProvider(fetchRequest)
+const smartContractClient = new ethers.Contract(
+  MINER_TO_PEERID_CONTRACT_ADDRESS,
+  MINER_TO_PEERID_CONTRACT_ABI,
+  provider,
+)
 
 /**
- * @param {object} options
- * @param {number} [options.maxAttempts]
- * @param {function} [options.rpcFn]
- * @returns {Promise<string>} The chain head Cid
+ * @param {string} minerId - The ID of the miner.
+ * @param {object} options - Options for the function.
+ * @param {number} options.maxAttempts - The maximum number of attempts to fetch the peer ID.
+ * @returns {Promise<string>} The peer ID of the miner.
  */
-async function getChainHead({ maxAttempts = 5, rpcFn } = {}) {
-  try {
-    const res = await retry(() => (rpcFn ?? rpc)('Filecoin.ChainHead'), {
-      // The maximum amount of attempts until failure.
-      maxAttempts,
-      // The initial and minimum amount of milliseconds between attempts.
-      minTimeout: 5_000,
-      // How much to backoff after each retry.
-      multiplier: 1.5,
-    })
-    return res.Cids
-  } catch (err) {
-    if (err.name === 'RetryError' && err.cause) {
-      // eslint-disable-next-line no-ex-assign
-      err = err.cause
-    }
-    err.message = `Cannot obtain chain head: ${err.message}`
-    throw err
-  }
-}
-
-/**
- * @param {string} minerId A miner actor id, e.g. `f0142637`
- * @param {object} options
- * @param {number} [options.maxAttempts]
- * @returns {Promise<string>} Miner's PeerId, e.g. `12D3KooWMsPmAA65yHAHgbxgh7CPkEctJHZMeM3rAvoW8CZKxtpG`
- */
-export async function getIndexProviderPeerId(
-  minerId,
-  { maxAttempts = 5, smartContract, rpcFn } = {},
-) {
-  try {
-    // Make a concurrent request to both sources: FilecoinMinerInfo and smart contract
-    const [minerInfoResult, contractResult] = await Promise.all([
-      getIndexProviderPeerIdFromFilecoinMinerInfo(minerId, { maxAttempts, rpcFn }),
-      getIndexProviderPeerIdFromSmartContract(minerId, { smartContract }),
-    ])
-    // Check contract result first
-    if (contractResult) {
-      console.log('Using PeerID from the smart contract.')
-      return contractResult
-    }
-
-    // Fall back to FilecoinMinerInfo result
-    if (minerInfoResult) {
-      console.log('Using PeerID from FilecoinMinerInfo.')
-      return minerInfoResult
-    }
-
-    // Handle the case where both failed
-    throw new Error(
-      `Failed to obtain Miner's Index Provider PeerID.\nSmartContract query result: ${contractResult}\nStateMinerInfo query result: ${minerInfoResult}`,
-    )
-  } catch (error) {
-    console.error('Error fetching PeerID:', error)
```

Comment:
Can you please confirm that with your new version, when the index-provider-peer-id lookup fails, Spark checker prints a descriptive error message?